### PR TITLE
add requirements for job_object_helper

### PR DESCRIPTION
**Pull request by BertKleewein**

### CODE REVIEW

File: devdoc/job_object_helper.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# job_object_help requirements
+
+`job_object_help` is a module that wraps the Windows JobObject API, primarily for the purpose of limiting the amount of memory and CPU the current process can consume.
+
+## Exposed API
+```c
+    MOCKABLE_FUNCTION(, int, job_object_helper_limit_resources, uint32_t, percent_pysical_memory, uint32_t, percent_cpu);
```
Comment (parth21999) [inline devdoc/job_object_helper.md:7]: >pysical [](http://example.com/codeflow?start=82&length=7)

buuurt

File: devdoc/job_object_helper.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+```c
+    MOCKABLE_FUNCTION(, int, job_object_helper_limit_resources, uint32_t, percent_pysical_memory, uint32_t, percent_cpu);
+```
+
+## job_object_helper_limit_resources
+```c
+    MOCKABLE_FUNCTION(, int, job_object_helper_limit_resources, uint32_t, percent_pysical_memory, uint32_t, percent_cpu);
+```
+
+`job_object_helper_limit_resources` is an API that uses the job objecgt API to limit the amount of pysical memory and CPU that the current process can consume.
```
Comment (parth21999) [inline devdoc/job_object_helper.md:15]: >objecgt [](http://example.com/codeflow?start=64&length=7)

buuurt

File: devdoc/job_object_helper.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+**SRS_JOB_OBJECT_HELPER_18_001: [** `percent_physical_memory` is allowed to be `0`, which means "do not limit physical memory usage". **]**
+
+**SRS_JOB_OBJECT_HELPER_18_002: [** `percent_cpu` is allowed to be `0`, which means "do not limit CPU usage". **]**
+
+**SRS_JOB_OBJECT_HELPER_18_003: [** if `percent_physical_memory` and `percent_cpu` are both `0`, `job_object_helper_limit_resources` shall succeed and return `0`. **]**
```
Comment (parth21999) [inline devdoc/job_object_helper.md:21]: > if  [](http://example.com/codeflow?start=35&length=4)

nit: capitalization

**Comment by parth21999** [comment]
This seems like it would be more at home in c-pal rather that c-util

File: devdoc/job_object_helper.md (hunk 1 above)
Comment (BertKleewein) [reply devdoc/job_object_helper.md:7]: Speeeling is not my strong soot,

**Comment by BertKleewein** [comment]
Good point.  moving code.

---
In reply to: [2356942323](https://github.com/Azure/c-util/pull/423#issuecomment-2356942323) [](http://example.com/codeflow?ancestors=2356942323)

**Pull request closed without merging**